File: DSA in Python/Basics_of_DSA/count_digit.py
"""Q) Input: 'X' = 2

Output: 1

As only one digit is ‘2’ present in ‘X’ so answer is ‘1’."""


def countdigit():
    n = "127868763"
    count = 0
    for i in n:
        count += 1
    print(count)


countdigit()

"""Using logarithmic method"""

from math import log10


def countdigit(n):
    if n == 0:
        return 1
    return int(log10(abs(n))) + 1


num = 12345678
result = countdigit(num)
print("Number of digits :", result)

File: JAVA/Loops in Java/count_digit.java
//count number of digit in a number

package com.company;

import java.util.Scanner;

public class count_digit {
    static void main() {
        Scanner sc=new Scanner(System.in);
        System.out.print("Enter a number =");
        int n=sc.nextInt();
        int count=0;
        do{
            count++;
            n/=10;
        }while(n!=0);
        System.out.println("Number of digits ="+count);
    }
}
